File: MiniDOMSIP/Config/DIALTP/DMDSIPConfigCursor.vb
Imports minidom.Databases

<Serializable>
Public NotInheritable Class DMDSIPConfigCursor
    Inherits DBObjectCursorBase(Of DMDSIPConfig)

    Private m_IDPostazione As New CCursorFieldObj(Of String)("IDPostazione")
    Private m_IDMacchina As New CCursorFieldObj(Of String)("IDMacchina")
    Private m_IDUtente As New CCursorFieldObj(Of String)("IDUtente")
    Private m_Attiva As New CCursorField(Of Boolean)("Attiva")
    Private m_DataInizio As New CCursorField(Of Date)("DataInizio")
    Private m_DataFine As New CCursorField(Of Date)("DataFine")
    Private m_Flags As New CCursorField(Of DMDSIPConfigFlags)("Flags")
    Private m_UploadServer As New CCursorFieldObj(Of String)("UploadServer")
    Private m_NotifyServer As New CCursorFieldObj(Of String)("NotifyServer")
    Private m_ServerName As New CCursorFieldObj(Of String)("ServerName")
    Private m_UserName As New CCursorFieldObj(Of String)("UserName")
    Private m_RemoveLogAfterNDays As New CCursorField(Of Integer)("RemoveLogAfterNDays")
    Private m_MinVersion As New CCursorFieldObj(Of String)("MinVersion")
    Private m_MaxVersion As New CCursorFieldObj(Of String)("MaxVersion")

    Public Sub New()
    End Sub

    Public ReadOnly Property MinVersion As CCursorFieldObj(Of String)
        Get
            Return Me.m_MinVersion
        End Get
    End Property

    Public ReadOnly Property MaxVersion As CCursorFieldObj(Of String)
        Get
            Return Me.m_MaxVersion
        End Get
    End Property

    Public ReadOnly Property RemoveLogAfterNDays As CCursorField(Of Integer)
        Get
            Return Me.m_RemoveLogAfterNDays
        End Get
    End Property

    Public ReadOnly Property ServerName As CCursorFieldObj(Of String)
        Get
            Return Me.m_ServerName
        End Get
    End Property

    Public ReadOnly Property UserName As CCursorFieldObj(Of String)
        Get
            Return Me.m_UserName
        End Get
    End Property

    Public ReadOnly Property UploadServer As CCursorFieldObj(Of String)
        Get
            Return Me.m_UploadServer
        End Get
    End Property

    Public ReadOnly Property NotifyServer As CCursorFieldObj(Of String)
        Get
            Return Me.m_NotifyServer
        End Get
    End Property

    Public ReadOnly Property IDPostazione As CCursorFieldObj(Of String)
        Get
            Return Me.m_IDPostazione
        End Get
    End Property

    Public ReadOnly Property IDMacchina As CCursorFieldObj(Of String)
        Get
            Return Me.m_IDMacchina
        End Get
    End Property

    Public ReadOnly Property IDUtente As CCursorFieldObj(Of String)
        Get
            Return Me.m_IDUtente
        End Get
    End Property

    Public ReadOnly Property Attiva As CCursorField(Of Boolean)
        Get
            Return Me.m_Attiva
        End Get
    End Property

    Public ReadOnly Property DataInizio As CCursorField(Of Date)
        Get
            Return Me.m_DataInizio
        End Get
    End Property

    Public ReadOnly Property DataFine As CCursorField(Of Date)
        Get
            Return Me.m_DataFine
        End Get
    End Property



    Public ReadOnly Property Flags As CCursorField(Of DMDSIPConfigFlags)
        Get
            Return Me.m_Flags
        End Get
    End Property

    Protected Overrides Function GetConnection() As CDBConnection
        Return DMDSIPApp.Database
    End Function

    Protected Overrides Function GetModule() As minidom.Sistema.CModule
        Return DMDSIPApp.Configs.Module
    End Function

    Public Overrides Function GetTableName() As String
        Return "tbl_DialTPConfig"
    End Function


End Class

File: MiniDOMSIP/Config/DIALTP/DMDSIPConfigCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;
namespace minidom.PBX
{

    /// <summary>
    /// Cursore di <see cref="DMDSIPConfig"/>
    /// </summary>
    [Serializable]
    public sealed class DMDSIPConfigCursor 
        : minidom.Databases.DBObjectCursorBase<DMDSIPConfig>
    {
        private DBCursorStringField m_IDPostazione = new DBCursorStringField("IDPostazione");
        private DBCursorStringField m_IDMacchina = new DBCursorStringField("IDMacchina");
        private DBCursorStringField m_IDUtente = new DBCursorStringField("IDUtente");
        private DBCursorField<bool> m_Attiva = new DBCursorField<bool>("Attiva");
        private DBCursorField<DateTime> m_DataInizio = new DBCursorField<DateTime>("DataInizio");
        private DBCursorField<DateTime> m_DataFine = new DBCursorField<DateTime>("DataFine");
        private DBCursorField<DMDSIPConfigFlags> m_Flags = new DBCursorField<DMDSIPConfigFlags>("Flags");
        private DBCursorStringField m_UploadServer = new DBCursorStringField("UploadServer");
        private DBCursorStringField m_NotifyServer = new DBCursorStringField("NotifyServer");
        private DBCursorStringField m_ServerName = new DBCursorStringField("ServerName");
        private DBCursorStringField m_UserName = new DBCursorStringField("UserName");
        private DBCursorField<int> m_RemoveLogAfterNDays = new DBCursorField<int>("RemoveLogAfterNDays");
        private DBCursorStringField m_MinVersion = new DBCursorStringField("MinVersion");
        private DBCursorStringField m_MaxVersion = new DBCursorStringField("MaxVersion");

        /// <summary>
        /// Costruttore
        /// </summary>
        public DMDSIPConfigCursor()
        {
        }

        /// <summary>
        /// MinVersion
        /// </summary>
        public DBCursorStringField MinVersion
        {
            get
            {
                return m_MinVersion;
            }
        }

        /// <summary>
        /// MaxVersion
        /// </summary>
        public DBCursorStringField MaxVersion
        {
            get
            {
                return m_MaxVersion;
            }
        }

        /// <summary>
        /// RemoveLogAfterNDays
        /// </summary>
        public DBCursorField<int> RemoveLogAfterNDays
        {
            get
            {
                return m_RemoveLogAfterNDays;
            }
        }

        /// <summary>
        /// ServerName
        /// </summary>
        public DBCursorStringField ServerName
        {
            get
            {
                return m_ServerName;
            }
        }

        /// <summary>
        /// UserName
        /// </summary>
        public DBCursorStringField UserName
        {
            get
            {
                return m_UserName;
            }
        }

        /// <summary>
        /// UploadServer
        /// </summary>
        public DBCursorStringField UploadServer
        {
            get
            {
                return m_UploadServer;
            }
        }

        /// <summary>
        /// NotifyServer
        /// </summary>
        public DBCursorStringField NotifyServer
        {
            get
            {
                return m_NotifyServer;
            }
        }

        /// <summary>
        /// IDPostazione
        /// </summary>
        public DBCursorStringField IDPostazione
        {
            get
            {
                return m_IDPostazione;
            }
        }

        /// <summary>
        /// IDMacchina
        /// </summary>
        public DBCursorStringField IDMacchina
        {
            get
            {
                return m_IDMacchina;
            }
        }

        /// <summary>
        /// IDUtente
        /// </summary>
        public DBCursorStringField IDUtente
        {
            get
            {
                return m_IDUtente;
            }
        }

        /// <summary>
        /// Attiva
        /// </summary>
        public DBCursorField<bool> Attiva
        {
            get
            {
                return m_Attiva;
            }
        }

        /// <summary>
        /// DataInizio
        /// </summary>
        public DBCursorField<DateTime> DataInizio
        {
            get
            {
                return m_DataInizio;
            }
        }

        /// <summary>
        /// DataFine
        /// </summary>
        public DBCursorField<DateTime> DataFine
        {
            get
            {
                return m_DataFine;
            }
        }

        /// <summary>
        /// Flags
        /// </summary>
        public DBCursorField<DMDSIPConfigFlags> Flags
        {
            get
            {
                return m_Flags;
            }
        }
 
        /// <summary>
        /// Repository
        /// </summary>
        /// <returns></returns>
        protected override CModulesClass GetModule()
        {
            return DMDSIPApp.Configs;
        }
         
    }
}